src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -252,6 +252,9 @@ private void OnAgentExit(Process process, Guid agentId)
                 case AgentExitCodes.UNABLE_TO_LOCATE_AGENCY:
                     errorMsg = "Remote test agent unable to locate agency process.";
                     break;
+                case AgentExitCodes.STACK_OVERFLOW_EXCEPTION:
+                    errorMsg = "Remote test agent was terminated due to a StackOverFlow exception in one of the tests.";
```

Comment:
```suggestion
                    errorMsg = "Remote test agent was terminated due to a stack overflow.";
```

How about something as simple as this? That way we're not saying any more than we definitely know.

Reply:
Yes it would be better. I will change it.